### ISSUE TITLE
Proposal: consistent repository labels

### ISSUE THREAD

**Issue by rowanmanning** [Financial-Times/origami#24]
Our repositories should have consistent labels, this will help with filtering the project board as well as automating issue or PR labelling on repos. This proposes which labels we should use across all of our repositories, as well as how we sync them.

## Labels

I've taken these from @chee's work in the proposals repo, but adding descriptions.

Label|Description|Color|Example
-----|-----|-----|-----
blocked|Work blocked by something else|`#CC0000` (crimson)|<img width="67" alt="label-blocked" src="https://user-images.githubusercontent.com/138944/76612724-c8c00480-6514-11ea-9dee-5e2344d31864.png">
breaking|Will require a major version bump|`#990F3D` (claret)|<img width="72" alt="label-breaking" src="https://user-images.githubusercontent.com/138944/76612740-d7a6b700-6514-11ea-9cb2-ecccfa30dc6b.png">
bug|Something isn't working|`#CC0000` (crimson)|<img width="41" alt="label-bug" src="https://user-images.githubusercontent.com/138944/76612741-d7a6b700-6514-11ea-885a-567d110b9e84.png">
dependencies|This is maintenance work relating to dependency bumps|`#FFEC1A` (lemon)|<img width="106" alt="label-dependencies" src="https://user-images.githubusercontent.com/138944/76756794-4fc3e580-677e-11ea-9364-c16c3231e3e4.png">
discussion|General discussion including support questions|`#FF7FAA` (candy)|<img width="84" alt="label-discussion" src="https://user-images.githubusercontent.com/138944/76612742-d83f4d80-6514-11ea-9a2e-1d4b3b3eec8d.png">
documentation|Improvements or additions to documentation|`#CCE6FF` (sky)|<img width="113" alt="label-documentation" src="https://user-images.githubusercontent.com/138944/76612743-d83f4d80-6514-11ea-8d29-cb9ef62751cf.png">
duplicate|This issue or pull request already exists|`#CCC1B7` (black-20)|<img width="75" alt="label-duplicate" src="https://user-images.githubusercontent.com/138944/76612745-d8d7e400-6514-11ea-83fa-66ce584708bf.png">
feature|New feature request|`#00994D` (jade)|<img width="62" alt="label-feature" src="https://user-images.githubusercontent.com/138944/76612746-d8d7e400-6514-11ea-910c-bc0a795ab7ae.png">
good starter issue|Good for newcomers|`#0D7680` (teal)|<img width="132" alt="label-good-starter-issue" src="https://user-images.githubusercontent.com/138944/76612748-d9707a80-6514-11ea-8a98-5caafa1f6fc2.png">
help wanted|We'd appreciate some help with this|`#96CC28` (wasabi)|<img width="94" alt="label-help-wanted" src="https://user-images.githubusercontent.com/138944/76747988-ef2eab80-6771-11ea-90f0-71ad15530cc9.png">
maintenance|Technical tasks that might make things better|`#FFEC1A` (lemon)|<img width="99" alt="label-maintenance" src="https://user-images.githubusercontent.com/138944/76612752-d9707a80-6514-11ea-9af3-d747bc12696d.png">
proposal|A proposed change which requires approval or discussion|`#262a33` (slate)|<img width="72" alt="label-proposal" src="https://user-images.githubusercontent.com/138944/76612754-da091100-6514-11ea-96b9-2602d529ea4f.png">
security|This includes a security patch|`#CC0000` (crimson)|<img width="68" alt="label-security" src="https://user-images.githubusercontent.com/138944/76756796-50f51280-677e-11ea-9cc9-e790ecf8f22a.png">
wontfix|This will not be worked on|`#CCC1B7` (black-20)|<img width="63" alt="label-wontfix" src="https://user-images.githubusercontent.com/138944/76612756-daa1a780-6514-11ea-8091-0509d9e94cac.png">

### Origami-Type Labels

We should also have labels for each `origamiType` so that we can filter issues by type (see #33). I'm unsure whether to include a prefix here for clarity, there are some types that might be used incorrectly, e.g. `config` – someone might add this when they're making a config change?

I'm leaning towards sticking to unprefixed. If we have a consistent way of syncing labels then we can change this later across the board without issue. Maybe at the point where something is labelled incorrectly?

Label|Description|Color|Example
-----|-----|-----|-----
[origamiType]|Relates to an Origami [origamiType]|`#593380` (velvet)|<img width="62" alt="label-service" src="https://user-images.githubusercontent.com/138944/76617875-19d4f600-651f-11ea-84bb-111122ce9203.png">
type:[origamiType]|Relates to an Origami [origamiType]|`#593380` (velvet)|<img width="95" alt="label-type-service" src="https://user-images.githubusercontent.com/138944/76617877-1b062300-651f-11ea-9355-262c3eeff496.png">

### Continuous Delivery Labels

We should also have labels for continuous delivery (see #30). These will be used to automate releases, it'd be nice if the labels themselves were automated too.

Label|Description|Color|Example
-----|-----|-----|-----
major|Add to a PR to trigger a MAJOR version bump when merged|`#990F3D` (claret)|<img width="52" alt="label-major" src="https://user-images.githubusercontent.com/138944/76637877-6af6e100-6543-11ea-9344-27c71860030b.png">
minor|Add to a PR to trigger a MINOR version bump when merged|`#00994D` (jade)|<img width="53" alt="label-minor" src="https://user-images.githubusercontent.com/138944/76637881-6c280e00-6543-11ea-811f-3cca4bc63cf9.png">
patch|Add to a PR to trigger a PATCH version bump when merged|`#CC0000` (crimson)|<img width="53" alt="label-patch" src="https://user-images.githubusercontent.com/138944/76637883-6cc0a480-6543-11ea-8fea-11012d88a2b7.png">

## Syncing

We should use [GitHub Label Sync](https://github.com/Financial-Times/github-label-sync) to do this. It does something that most other automated labelling software doesn't: renaming existing labels rather than removing them all and creating new ones. This is important to us as we don't want to lose historical labels.

We should use GitHub Label Sync via a new GitHub action named `origami-labels`. This action would be versioned and contain JSON which describes our set of labels.

**Comment by notlee** [comment]
I don't think we need "feature" for use on PRs -- we may have (major,minor,patch)[https://github.com/Financial-Times/origami/issues/30] soon and a minor is effectively feature.

Could we have "feature request" though for use on issues?

**Comment by notlee** [comment]
We should also add an [origamiType label](https://github.com/Financial-Times/origami/issues/33)! Maybe `origami-type: {type}`? 

Is it possible to apply the correct type label depending on `origami.json` simply or will we better off to sync all type labels to all repos, and just use the correct one on issues?

**Comment by rowanmanning** [comment]
What if I change the description of `feature` to `New feature request`?

**Comment by rowanmanning** [comment]
Ah yeah I missed this https://github.com/Financial-Times/origami/issues/33! I'll add some more suggestions for `component`, `service`, etc

**Comment by rowanmanning** [comment]
I also just added labels for `major`, `minor`, `patch`. I used the same colours as `breaking`, `feature`, `bug` but not sure if this seems right. Open to suggestions

**Comment by rowanmanning** [comment]
I've got a GitHub action ready to go! It was surprisingly quick and easy 🙂
https://github.com/Financial-Times/origami-labels

Will wait for more feedback on this proposal before rolling it out across our repos.

**Comment by rowanmanning** [comment]
I've noticed that Dependabot seems to add labels to repos in some cases. Looking at Polyfill Library as an example it has `dependencies` and `security` labels. Do we want to include these?

**Comment by rowanmanning** [comment]
Hmm, polyfill also has a `released` label. What's this used for @JakeChampion? Do we need to make sure it isn't overridden by the label syncing?

**Comment by notlee** [comment]
>I'm leaning towards sticking to unprefixed. If we have a consistent way of syncing labels then we can change this later across the board without issue. Maybe at the point where something is labelled incorrectly?

👍

**Comment by rowanmanning** [comment]
Origami website has a `todo` label. Is this also needed, @chee? And @notlee what do we think about the `UXD` label?

Sorry for all the questions, just going through and checking all of our labels.

**Comment by notlee** [comment]
Hmm, let's remove the `UXD` label for now. I think even if something is blocked by UXD or requires UXD input, it's not a very useful label. As most of the Product Design team don't use Github much currently.

**Comment by chee** [comment]
i prefer unprefixed for the origami type :)) (and probably ignore it if the origamiType is null lol)

**Comment by JakeChampion** [comment]
> I've noticed that Dependabot seems to add labels to repos in some cases. Looking at Polyfill Library as an example it has `dependencies` and `security` labels. Do we want to include these?

Those were added by Dependabot, so I assume Dependabot would add them again if they got removed?

> Hmm, polyfill also has `released` and `help wanted` labels. What are these used for @JakeChampion? Do we need to make sure they aren't overridden by the label syncing?

I would like to keep `help wanted` and `good first issue` because they help power the contribute page --> https://github.com/Financial-Times/polyfill-library/contribute

`released` is powered by a tool I use on some repos, I will be not using that tool in the future, so no need to keep it.

**Comment by chee** [comment]
let's include the dependabot ones because then we can add our own descriptions to them and force them to be o-colored ^_^!

@rowanmanning the origami website has a `todo` label because it was the proposals board for a while. at the time it seemed like we needed something for tasks that just needed to get done (such as "Add `origami.json` to the repos from which it is missing").

**Comment by JakeChampion** [comment]
For the record,

I am in favour of this proposal :-)

**Comment by rowanmanning** [comment]
OK I think I've addressed all the outstanding questions, thanks everyone

**Comment by chee** [comment]
there's one more thing which is this: https://github.com/Financial-Times/origami-website/blob/master/_specification-v1/components.md#component-feature-deprecation

> An issue with label `type: breaking` must be created as a reminder to remove deprecated code upon a future major release.

we can change this to a label of `breaking`

**Comment by rowanmanning** [comment]
This is all done now!

**Comment by chee** [comment]
yay!!
